deployments/quick-start/setup.ps1

```diff
@@ -0,0 +1,203 @@
+    if ($vault_pod -ne $null) {
+        Write-Host "Vault pod name: $($vault_pod.metadata.name)"
+    }
+    else {
+        Write-Host "Vault not started. Waiting 1 second."
+        Start-Sleep -Seconds 1
+        $vault_pod = Find-VaultPod
+    }
+}
+
+# And if the Pod isn't running, let's wait for it to become running.
+while ($vault_pod.status.phase -ne "Running") {
+    Write-Host "$($vault_pod.metadata.name) is $($vault_pod.status.phase). Waiting 1 second for it to become Running."
+    Start-Sleep -Seconds 1
+    $vault_pod = Find-VaultPod
+}
+
+
+# Run a background job to set up Port Forwarding to Vault
+$pf_job = "kubectl port-forward $($vault_pod.metadata.name) 8200"
+Write-Host "This can occasionally hang. Pressing an arrow key with the kubectl window open will let the process continue."
+$proxy = Start-Process -FilePath powershell.exe -ArgumentList "-NoProfile -ExecutionPolicy Bypass -command $pf_job" -PassThru
+
+# Set the Vault Address environment variable
+[Environment]::SetEnvironmentVariable("VAULT_ADDR", "http://127.0.0.1:8200", "Process")
+
+# Authenticat to Vault
+vault auth vault-root-token
+
+# Set up the Root Certificate Authority
+# Create a Root CA that expires in 10 years:
+vault mount -path=root-ca -max-lease-ttl=87600h pki
+
+# Generate the root certificate:
+vault write root-ca/root/generate/internal common_name="Root CA" ttl=87600h exclude_cn_from_sans=true
+
+# Set up the URLs:
+vault write root-ca/config/urls issuing_certificates="http://vault:8200/v1/root-ca/ca" crl_distribution_points="http://vault:8200/v1/root-ca/crl"
+
+# Create the Intermediate Certificate Authority
+# Create the Intermediate CA that expires in 5 years:
+vault mount -path=intermediate-ca -max-lease-ttl=43800h pki
+
+# Generate a Certificate Signing Request:
+$intermediate_csr_response = vault write -format=json intermediate-ca/intermediate/generate/internal common_name="Intermediate CA" ttl=43800h exclude_cn_from_sans=true
+
+$intermediate_csr = "$($intermediate_csr_response)" | ConvertFrom-Json
+
+$csr_data = $intermediate_csr.data.csr
+
+# Ask the Root to sign it:
+$intermediate_cert_response = vault write -format=json root-ca/root/sign-intermediate csr=$csr_data use_csr_values=true exclude_cn_from_sans=true
+
+$signed_cert = "$($intermediate_cert_response)" | ConvertFrom-Json
+$signed_data = $signed_cert.data.certificate
+
+# Send the signed certificate back to Vault:
+vault write intermediate-ca/intermediate/set-signed certificate=$signed_data
+
+# Set up URLs:
+vault write intermediate-ca/config/urls issuing_certificates="http://vault:8200/v1/intermediate-ca/ca" crl_distribution_points="http://vault:8200/v1/intermediate-ca/crl"
+
+# Enable the AppRole backend
+vault auth-enable approle
+
+# Create a role to allow Kubernetes-Vault to generate certificates:
+vault write intermediate-ca/roles/kubernetes-vault allow_any_name=true max_ttl="24h"
+
+# Inspect the policy file deployments/quick-start/policy-kubernetes-vault.hcl
```

Comment:
This comment can be removed.

Reply:
That's copied directly from the setup.sh